core/sql/generator/GenRelScan.cpp

```diff
@@ -302,7 +302,7 @@ int HbaseAccess::createAsciiColAndCastExpr2(Generator * generator,
   asciiValue = new (h) NATypeToItem(newGivenType->newCopy(h));
   castValue = new(h) Cast(asciiValue, newGivenType); 
 
-  if (HbaseAccess::isEncodingNeededForSerialization(colNode))
+  if ((!alignedFormat) && HbaseAccess::isEncodingNeededForSerialization(colNode))
```

Comment:
This comment is about line 264, the NABoolean HbaseAccess::isEncodingNeededForSerialization(ItemExpr * colNode) method. I could not add a comment  there.

Is this method expected to return the correct answer now if it is called on an Index object which is aligned, but whose parent table is not aligned? I would expect the answer to be  YES, and therefore the additional change in line 305 is not necessary. If that is not the case, then some kind of comment on line 296 starting that it cannot be used directly on aligned indexes will be good.